src/csharp/Microsoft.Spark.Worker/DaemonWorker.cs

```diff
@@ -51,6 +51,8 @@ internal DaemonWorker(Version version)
             _version = version;
         }
 
+        internal ConcurrentDictionary<int, TaskRunner> GetTaskRunners() => _taskRunners;
```

Comment:
Instead of exposing the whole structure, shall we return `CurrentNumTaskRunners` since that should be sufficient for your tests.